pkg/operator/ceph/object/spec.go

```diff
@@ -232,21 +236,25 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
 		mount := v1.VolumeMount{Name: certVolumeName, MountPath: certDir, ReadOnly: true}
 		container.VolumeMounts = append(container.VolumeMounts, mount)
 	}
-	if c.clusterSpec.Security.KeyManagementService.IsEnabled() {
+	kmsEnabled, err := c.CheckRGWKMS()
+	if err != nil {
+		return container
```

Comment:
Return an empty struct, we don't want to run a half-complete container, the caller will check for an empty struct. To make it easier we should return a pointer of `v1.Container` probably...